# List of all nodes in the structure
# Values correspond to x,y,z in meters
node_positions:
#  - position: [1.5, -1.3, 2.0]
#  - position: [2.0, 0.2, 2.5]
#  - position: [1.0, -1.0, -0.2]
  - position: [0.0, 0.0, 0.0]
  - position: [0.0, 1.0, 1.0]
#  - position: [1.0, -1.0, -0.2]

#  - position: [5.416697, 3.93552, 1.21057]
#  - position: [4.675475, 0.08346, 1.21057]
#  - position: [8.382068, 1.367573, 1.21057]
#  - position: [7.640729, 0.083573, 1.210573]
#  - position: [3.934168, 1.367478, 1.210573]
#  - position: [6.899343, 3.935502, 1.210573]
#  - position: [3.934067, 3.935517, 1.210594]
#  - position: [8.382077, 3.935593, 1.210594]
#  - position: [6.158123, 0.083468, 1.210597]
#  - position: [7.640722, 2.651603, 1.210597]
#  - position: [6.899428, -1.200548, 1.210597]
#  - position: [4.675355, 2.651495, 1.210599]
#  - position: [9.123368, 2.651563, 1.210603]
#  - position: [5.416789, -1.200506, 1.210603]
#  - position: [3.192699, 2.651549, 1.210605]
#  - position: [5.416782, 1.367488, 1.210624]
#  - position: [6.899413, 1.367549, 1.210624]
#  - position: [6.158045, 2.651515, 1.210624]
#  - position: [3.192794, 0.939451, 2.421154]
#  - position: [6.899348, 4.791563, 2.421154]
#  - position: [8.382098, -0.344462, 2.421154]
#  - position: [7.640731, 3.507586, 2.421159]
#  - position: [6.89943, -0.344524, 2.421159]
#  - position: [3.934056, 2.223516, 2.421159]
#  - position: [5.416708, 4.791548, 2.421159]
#  - position: [9.123405, 0.939549, 2.421159]
#  - position: [3.934127, -0.344545, 2.421159]
#  - position: [3.934071, 4.791549, 2.42116]
#  - position: [8.382079, 4.791622, 2.42116]
#  - position: [4.675486, 0.939478, 2.421162]
#  - position: [6.158025, 3.507502, 2.421162]
#  - position: [7.640729, 0.939573, 2.421162]
#  - position: [2.451349, 2.223533, 2.421163]
#  - position: [8.382046, 2.22361, 2.421165]
#  - position: [4.675354, 3.507502, 2.421165]
#  - position: [5.416817, -0.344533, 2.421165]
#  - position: [4.675454, -1.628508, 2.421169]
#  - position: [3.192719, 3.507508, 2.421171]
#  - position: [9.123365, 3.507607, 2.421171]
#  - position: [9.864709, 2.22355, 2.421177]
#  - position: [6.158106, 0.939505, 2.421178]
#  - position: [6.899384, 2.223557, 2.421179]
#  - position: [5.416724, 2.223496, 2.421182]
#  - position: [7.640755, -1.628548, 2.421182]
#  - position: [6.158124, -1.628532, 2.421187]
#  - position: [6.158029, 4.363545, 3.631738]
#  - position: [3.934128, 0.51146, 3.631738]
#  - position: [8.382083, 0.511552, 3.631761]
#  - position: [6.158068, 1.795516, 3.631764]
#  - position: [7.640774, -0.772525, 3.631765]
#  - position: [3.192735, 1.795492, 3.631765]
#  - position: [7.640722, 4.363598, 3.631765]
#  - position: [6.899397, 3.079564, 3.631765]
#  - position: [4.675403, 1.795503, 3.631765]
#  - position: [6.89943, 0.511498, 3.631765]
#  - position: [5.416701, 3.079489, 3.631768]
#  - position: [5.416816, 0.511487, 3.631768]
#  - position: [7.640713, 1.795588, 3.631768]
#  - position: [4.675374, 4.363514, 3.63177]
#  - position: [4.675482, -0.772533, 3.63177]
#  - position: [9.123375, 1.795584, 3.63177]
#  - position: [2.451422, 3.079547, 3.631775]
#  - position: [9.123389, 4.36357, 3.631775]
#  - position: [3.192759, 4.363559, 3.631775]
#  - position: [9.864706, 3.079547, 3.631775]
#  - position: [6.158135, -0.772503, 3.631779]
#  - position: [8.382018, 3.079599, 3.631779]
#  - position: [3.934058, 3.079487, 3.631779]
#  - position: [6.158077, 2.651511, 4.842356]
#  - position: [9.864722, 1.367563, 4.842356]
#  - position: [4.67538, 5.21954, 4.842356]
#  - position: [3.934142, -1.20055, 4.842356]
#  - position: [5.41676, 1.367507, 4.842356]
#  - position: [6.899396, 1.367521, 4.842364]
#  - position: [9.123418, 0.083556, 4.84237]
#  - position: [7.640767, 0.083518, 4.84238]
#  - position: [8.382049, 1.367566, 4.842383]
#  - position: [6.899379, 3.935568, 4.842386]
#  - position: [4.675484, 0.083488, 4.842386]
#  - position: [5.416719, 3.935499, 4.842386]
#  - position: [3.934094, 1.367475, 4.842386]
#  - position: [7.640722, 5.219606, 4.842417]
#  - position: [2.451413, 1.367477, 4.842417]
#  - position: [8.382109, -1.200531, 4.842417]
#  - position: [6.158039, 0.939514, 6.052924]
#  - position: [5.416787, 2.223533, 6.052924]
#  - position: [6.899406, 2.223466, 6.052924]
#  - position: [6.158047, 3.507535, 6.052931]
#  - position: [4.675445, 0.939471, 6.052931]
#  - position: [9.123401, 0.939553, 6.052937]
#  - position: [3.934134, -0.344543, 6.052937]
#  - position: [5.416708, 4.791542, 6.052937]
#  - position: [7.640751, 0.939549, 6.052943]
#  - position: [8.382085, -0.344466, 6.052951]
#  - position: [3.1928, 0.939466, 6.052951]
#  - position: [6.899359, 4.791553, 6.052951]

# List of all beams in the structure
# List correspond to pairs of indices of nodes
# nodes: [0, 1] constructs a beam from node 0 to node 1
beams:
  - nodes: [0, 1]
#  - nodes: [1, 2]

#  - nodes: [61,32]
#  - nodes: [32,50]
#  - nodes: [37,61]
#  - nodes: [50,82]
#  - nodes: [37,14]
#  - nodes: [63,37]
#  - nodes: [18,50]
#  - nodes: [14,23]
#  - nodes: [50,23]
#  - nodes: [67,37]
#  - nodes: [6,37]
#  - nodes: [27,63]
#  - nodes: [80,50]
#  - nodes: [18,46]
#  - nodes: [80,94]
#  - nodes: [18,4]
#  - nodes: [67,23]
#  - nodes: [27,6]
#  - nodes: [26,46]
#  - nodes: [4,23]
#  - nodes: [23,11]
#  - nodes: [23,53]
#  - nodes: [58,27]
#  - nodes: [34,6]
#  - nodes: [67,34]
#  - nodes: [53,80]
#  - nodes: [26,59]
#  - nodes: [88,80]
#  - nodes: [90,78]
#  - nodes: [71,59]
#  - nodes: [26,1]
#  - nodes: [4,29]
#  - nodes: [46,29]
#  - nodes: [34,58]
#  - nodes: [11,34]
#  - nodes: [70,58]
#  - nodes: [78,88]
#  - nodes: [1,29]
#  - nodes: [59,36]
#  - nodes: [59,78]
#  - nodes: [29,53]
#  - nodes: [53,42]
#  - nodes: [55,34]
#  - nodes: [0,34]
#  - nodes: [58,79]
#  - nodes: [24,58]
#  - nodes: [42,11]
#  - nodes: [88,72]
#  - nodes: [72,53]
#  - nodes: [29,15]
#  - nodes: [36,13]
#  - nodes: [1,35]
#  - nodes: [78,56]
#  - nodes: [35,59]
#  - nodes: [29,56]
#  - nodes: [42,55]
#  - nodes: [24,0]
#  - nodes: [91,79]
#  - nodes: [79,55]
#  - nodes: [85,72]
#  - nodes: [15,42]
#  - nodes: [56,72]
#  - nodes: [35,13]
#  - nodes: [56,35]
#  - nodes: [0,30]
#  - nodes: [30,55]
#  - nodes: [79,87]
#  - nodes: [72,84]
#  - nodes: [24,45]
#  - nodes: [17,42]
#  - nodes: [55,68]
#  - nodes: [72,48]
#  - nodes: [48,42]
#  - nodes: [13,44]
#  - nodes: [15,40]
#  - nodes: [68,85]
#  - nodes: [65,35]
#  - nodes: [8,35]
#  - nodes: [40,56]
#  - nodes: [45,30]
#  - nodes: [30,17]
#  - nodes: [8,40]
#  - nodes: [40,48]
#  - nodes: [48,68]
#  - nodes: [68,87]
#  - nodes: [65,44]
#  - nodes: [5,30]
#  - nodes: [30,52]
#  - nodes: [19,45]
#  - nodes: [87,77]
#  - nodes: [68,52]
#  - nodes: [84,73]
#  - nodes: [17,41]
#  - nodes: [48,41]
#  - nodes: [86,68]
#  - nodes: [44,10]
#  - nodes: [73,48]
#  - nodes: [16,40]
#  - nodes: [65,22]
#  - nodes: [54,40]
#  - nodes: [22,8]
#  - nodes: [19,5]
#  - nodes: [77,95]
#  - nodes: [52,77]
#  - nodes: [73,86]
#  - nodes: [16,41]
#  - nodes: [52,41]
#  - nodes: [73,54]
#  - nodes: [22,54]
#  - nodes: [10,22]
#  - nodes: [5,21]
#  - nodes: [19,51]
#  - nodes: [41,57]
#  - nodes: [77,51]
#  - nodes: [41,9]
#  - nodes: [57,73]
#  - nodes: [31,54]
#  - nodes: [3,22]
#  - nodes: [92,73]
#  - nodes: [31,16]
#  - nodes: [49,22]
#  - nodes: [21,52]
#  - nodes: [10,43]
#  - nodes: [54,75]
#  - nodes: [51,81]
#  - nodes: [31,57]
#  - nodes: [3,31]
#  - nodes: [43,49]
#  - nodes: [92,75]
#  - nodes: [21,9]
#  - nodes: [51,21]
#  - nodes: [75,49]
#  - nodes: [66,21]
#  - nodes: [9,33]
#  - nodes: [76,57]
#  - nodes: [57,33]
#  - nodes: [47,31]
#  - nodes: [20,3]
#  - nodes: [2,31]
#  - nodes: [76,92]
#  - nodes: [47,75]
#  - nodes: [75,93]
#  - nodes: [7,21]
#  - nodes: [49,83]
#  - nodes: [20,49]
#  - nodes: [28,51]
#  - nodes: [66,33]
#  - nodes: [28,7]
#  - nodes: [2,33]
#  - nodes: [47,76]
#  - nodes: [20,47]
#  - nodes: [66,38]
#  - nodes: [33,12]
#  - nodes: [60,76]
#  - nodes: [33,60]
#  - nodes: [62,28]
#  - nodes: [38,7]
#  - nodes: [89,76]
#  - nodes: [25,2]
#  - nodes: [25,47]
#  - nodes: [74,47]
#  - nodes: [38,62]
#  - nodes: [12,38]
#  - nodes: [25,60]
#  - nodes: [60,39]
#  - nodes: [64,38]
#  - nodes: [39,12]
#  - nodes: [69,60]
#  - nodes: [39,64]

# Which beams are sensors
# nodes: [0, 1] constructs a sensor from node 0 to node 1
sensors:
  - nodes: [0, 1]
#  - nodes: [1, 2]
#  - nodes: [4, 6]
#  - nodes: [1, 3]
#  - nodes: [4, 7]

# Which beams have light strips
# nodes: [0, 1] constructs a light strip from node 0 to node 1
light_beams:
  - nodes: [0, 1]
#  - nodes: [3, 7]
#  - nodes: [4, 5]
#  - nodes: [5, 6]
#  - nodes: [6, 7]

# Which nodes have light strips wrapped around
# [0, 2, 5] corresponds to light nodes at node 0, 2, and 5
# light_nodes: [8, 9, 11, 15, 16, 17, 21, 22, 23, 29, 30, 31, 33, 34, 35, 40, 41, 42, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 65, 66, 67, 68, 72, 73, 75, 76, 77, 78, 79, 80, 84, 85, 86, 87, 88, 92]
light_nodes: [0]

num_leds_of_each_light_beam: [10]
num_leds_of_each_light_node: [5]

# This list contains the arduino id to which this 
# light node is hooked up
# [2, 1, 2] means
# light node 0 is hooked up with arduino 2
# light node 1 is hooked up with arduino 1
# light node 2 is hooked up with arduino 2
# Note: Each node light must be hooked up to exactly
# one arduino. Thus, number of light_node_connections
# must equal the number of light_nodes specified.
# light_node_connections: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
light_node_connections: [0]
# This list contains the arduino id to which this 
# light beam is hooked up
# [2, 1, 2] means
# light beam 0 is hooked up with arduino 2
# light beam 1 is hooked up with arduino 1
# light beam 2 is hooked up with arduino 2
# Note: Each beam light must be hooked up to exactly
# one arduino. Thus, number of light_beam_connections
# must equal the number of light_beams specified.
light_beam_connections: [0]
# This list contains the arduino id to which this 
# sensor is hooked up
# [2, 1, 2] means
# sensor 0 is hooked up with arduino 2
# sensor 1 is hooked up with arduino 1
# sensor 2 is hooked up with arduino 2
# Note: Each sensor must be hooked up to exactly
# one arduino. Thus, number of sensor_connections
# must equal the number of sensors specified.
sensor_connections: [0]

# Size of the nodes and color
nodes_size:
  x: 0.07
  y: 0.07
  z: 0.07
nodes_color: [0.8, 0.8, 0.8, 1.0] # r,g,b,a

# Size of the light nodes and color
light_nodes_size:
  x: 0.1
  y: 0.1
  z: 0.1
light_nodes_color: [0.9, 0.9, 0.9, 1.0] # r,g,b,a

# Size of the beams and color
beams_size:
  x: 0.03
  y: 0.03
  z: 1.0
beams_color: [0.4, 0.4, 0.4, 1.0] # r,g,b,a

# Size of the light beams and color
light_beams_size:
  x: 0.08
  y: 0.08
  z: 0.8
light_beams_color: [0.8, 0.4, 0.4, 1.0] # r,g,b,a

# Size of the sensors and color
sensors_size:
  x: 0.04
  y: 0.04
  z: 0.9
sensors_color: [0.2, 0.2, 0.9, 0.9] # r,g,b,a

